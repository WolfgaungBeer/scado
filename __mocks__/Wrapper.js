import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Theme, defaultTheme } from '../scado';

const mockStore = configureStore();
const store = mockStore({
    scado: {
        theme: defaultTheme,
        toasts: [
            { id: '1', type: 'success', content: 'Text', onClose: () => {} },
            { id: '2', type: 'error', content: 'Text', onClose: () => {} },
        ],
    },
});


const Wrapper = ({ children }) => ( // eslint-disable-line
    <Provider store={store}>
        <Theme>{children}</Theme>
    </Provider>
);

export default Wrapper;
