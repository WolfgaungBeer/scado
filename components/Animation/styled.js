import styled from 'styled-components';

export const AnimationWrapper = styled.div`
    display: block;
`;

AnimationWrapper.displayName = 'AnimationWrapper';
