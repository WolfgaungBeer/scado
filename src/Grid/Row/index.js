import React, { Children, cloneElement } from 'react';
import { string, node } from 'prop-types';

const propTypes = {
    gutter: string,
    children: node,
};

const defaultProps = {
    gutter: undefined,
    children: undefined,
};

const Row = ({ gutter, children }) => {
    const gutters = gutter ? gutter.split(/ /) : [];
    return (
        <div>
            {gutter ? Children.map(children, child =>
                cloneElement(child, { gutterV: gutters[0], gutterH: gutters[1] }),
            ) : children}
        </div>
    );
};

Row.propTypes = propTypes;
Row.defaultProps = defaultProps;

export default Row;
