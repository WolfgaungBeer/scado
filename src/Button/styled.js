import styled from 'styled-components';
import { getBackgroundColor, getBorder, getShadow, getFontColor } from './helpers';

export const StyledButton = styled.button`
    width: 100%;
    cursor: ${props => (props.disabled ? 'not-allowed' : 'pointer')};
    background-color: ${getBackgroundColor()};
    ${getBorder()};
    ${getShadow()};
    color: ${getFontColor()};
    font-size: ${props => props.theme.scale[props.scale]};
    padding: 0.5rem 1rem;
    &:focus {
        outline: 0;
    }
`;
