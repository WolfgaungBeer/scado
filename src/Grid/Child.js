import styled from 'styled-components';
import { string, node } from 'prop-types';

const propTypes = {
    area: string,
    className: string,
    children: node,
};

const defaultProps = {
    area: undefined,
    className: undefined,
    children: undefined,
};

const Child = styled.div`
    grid-area: ${props => props.area};
`;

Child.propTypes = propTypes;
Child.defaultProps = defaultProps;

export default Child;
