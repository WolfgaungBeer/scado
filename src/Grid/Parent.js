import styled from 'styled-components';
import { bool, string, node } from 'prop-types';

const propTypes = {
    inline: bool,
    columns: string,
    rows: string,
    areas: string,
    columnGap: string,
    rowGap: string,
    className: string,
    children: node,
};

const defaultProps = {
    inline: undefined,
    columns: undefined,
    rows: undefined,
    areas: undefined,
    columnGap: undefined,
    rowGap: undefined,
    className: undefined,
    children: undefined,
};

const Parent = styled.div`
    display: ${props => (props.inline ? 'inline-grid' : 'grid')};
    grid-template-columns: ${props => props.columns};
    grid-template-rows: ${props => props.rows};
    grid-template-areas: ${props => props.areas};
    grid-column-gap: ${props => props.columnGap};
    grid-row-gap: ${props => props.rowGap};
    width: 100%;
    height: 100%;
`;

Parent.propTypes = propTypes;
Parent.defaultProps = defaultProps;

export default Parent;
