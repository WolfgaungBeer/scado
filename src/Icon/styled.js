import styled, { keyframes, css } from 'styled-components';

const rotationKeyframes = keyframes`
    from { transform: rotate(0deg); }
    to { transform: rotate(360deg); }
`;

export const StyledIcon = styled.i`
    cursor: ${props => props.onClick && 'pointer'};
    vertical-align: -16%;
    padding: 0 0.25rem;
    color: ${props => props.theme.color[props.color] || 'inherit'};
    font-size: ${props => props.theme.scale[props.scale] || 'inherit'};
    ${(props) => {
        if (props.loading) {
            return css`
                display: inline-block;
                animation: ${rotationKeyframes} 0.85s linear infinite;
            `;
        }
        return undefined;
    }};
`;
