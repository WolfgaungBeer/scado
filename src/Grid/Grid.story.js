import React from 'react';
import styled from 'styled-components';
import { storiesOf } from '@storybook/react';
import Grid from './';

const gridProps = {
    columns: 'auto',
    rows: '4rem 20rem',
    areas: '"header header" "sidebar body"',
    columnGap: '1rem',
    rowGap: '1rem',
};

const Panel = styled.div`
    width: 100%;
    height: 100%;
    background-color: ${props => props.theme.color.primary};
`;


storiesOf('Grid', module)
    .add('default', () => (
        <Grid.Parent {...gridProps}>
            <Grid.Child area="header"><Panel>Header</Panel></Grid.Child>
            <Grid.Child area="sidebar"><Panel>Sidebar</Panel></Grid.Child>
            <Grid.Child area="body"><Panel>Body</Panel></Grid.Child>
        </Grid.Parent>
    ))
    .add('inline', () => (
        <Grid.Parent {...gridProps} inline>
            <Grid.Child area="header"><Panel>Header</Panel></Grid.Child>
            <Grid.Child area="sidebar"><Panel>Sidebar</Panel></Grid.Child>
            <Grid.Child area="body"><Panel>Body</Panel></Grid.Child>
        </Grid.Parent>
    ));
