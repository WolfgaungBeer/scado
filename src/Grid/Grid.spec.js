import React from 'react';
import styled from 'styled-components';
import renderer from 'react-test-renderer';
import Wrapper from '../../__mocks__/Wrapper';
import Grid from './';

const gridProps = {
    columns: 'auto',
    rows: '4rem 20rem',
    areas: '"header header" "sidebar body"',
    columnGap: '1rem',
    rowGap: '1rem',
};

const Panel = styled.div`
    width: 100%;
    height: 100%;
    background-color: ${props => props.theme.color.primary};
`;

it('renders the Flex Component correctly', () => {
    const tree1 = renderer.create(
        <Wrapper>
            <Grid.Parent {...gridProps}>
                <Grid.Child area="header"><Panel>Header</Panel></Grid.Child>
                <Grid.Child area="sidebar"><Panel>Sidebar</Panel></Grid.Child>
                <Grid.Child area="body"><Panel>Body</Panel></Grid.Child>
            </Grid.Parent>
        </Wrapper>,
    ).toJSON();
    expect(tree1).toMatchSnapshot();

    const tree2 = renderer.create(
        <Wrapper>
            <Grid.Parent {...gridProps} inline>
                <Grid.Child area="header"><Panel>Header</Panel></Grid.Child>
                <Grid.Child area="sidebar"><Panel>Sidebar</Panel></Grid.Child>
                <Grid.Child area="body"><Panel>Body</Panel></Grid.Child>
            </Grid.Parent>
        </Wrapper>,
    ).toJSON();
    expect(tree2).toMatchSnapshot();
});
