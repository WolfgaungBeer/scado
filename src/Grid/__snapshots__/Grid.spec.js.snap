// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Flex Component correctly 1`] = `
<div
  className="sc-htpNat gPNHok"
  rows="4rem 20rem"
>
  <div
    className="sc-bxivhb fJzdIF"
  >
    <div
      className="sc-eNQAEJ fcVFnE"
    >
      Header
    </div>
  </div>
  <div
    className="sc-bxivhb dwDvly"
  >
    <div
      className="sc-eNQAEJ fcVFnE"
    >
      Sidebar
    </div>
  </div>
  <div
    className="sc-bxivhb kiaMgL"
  >
    <div
      className="sc-eNQAEJ fcVFnE"
    >
      Body
    </div>
  </div>
</div>
`;

exports[`renders the Flex Component correctly 2`] = `
<div
  className="sc-htpNat iIinON"
  rows="4rem 20rem"
>
  <div
    className="sc-bxivhb fJzdIF"
  >
    <div
      className="sc-eNQAEJ fcVFnE"
    >
      Header
    </div>
  </div>
  <div
    className="sc-bxivhb dwDvly"
  >
    <div
      className="sc-eNQAEJ fcVFnE"
    >
      Sidebar
    </div>
  </div>
  <div
    className="sc-bxivhb kiaMgL"
  >
    <div
      className="sc-eNQAEJ fcVFnE"
    >
      Body
    </div>
  </div>
</div>
`;
